client/src/App.jsx

```diff
@@ -0,0 +1,145 @@
+import { userService } from './services/userService.js';
+import { Formik, Form, Field } from 'formik';
+import { useContext, useEffect } from 'react';
+import { Context } from './context/context.jsx';
+import { ChatBlock } from './pages/ChatsBlock.jsx';
+
+function nicknameValidation(nickname) {
+  if (!nickname || nickname.trim().length === 0) {
+    return 'Nickname required';
+  }
+
+  if (nickname.trim().length < 2) {
+    return 'Nickname is too short';
+  }
+
+  if (nickname.trim().includes(' ')) {
+    return 'One word reqiured';
+  }
+}
+
+export const App = () => {
+  const { user, setUser, setAnyError, setAllChats } = useContext(Context);
+
+  useEffect(() => {
+    const storedUser = localStorage.getItem('nickname');
+    setUser(storedUser);
```

Comment:
Consider checking if `storedUser` is not null before calling `setUser(storedUser)`. This will prevent setting a null value to the user state if no nickname is stored in localStorage.